.circleci/config.yml

```diff
@@ -28,6 +28,17 @@ commands:
             ACTUAL_SIGNATURE=$(php -r "echo hash_file('sha384', 'composer-setup.php');")
             [[ "$EXPECTED_SIGNATURE" == "$ACTUAL_SIGNATURE" ]] && sudo php composer-setup.php --install-dir=/bin --filename=composer || exit 1
 
+  conditionally-skip:
+    description: "Skip if the current commit has no change in relevant directories"
+    parameters:
+      directories:
+        type: string
+    steps:
+      - run:
+          command: |
+            CHANGED_FILES="$( git diff HEAD^ HEAD --name-only --diff-filter=ACMRT << parameters.directories >> )"
+            if [ -z "$CHANGED_FILES" ] && [ -z "$CIRCLE_TAG" ]; then echo "No change to relevant file, exiting job" && circleci-agent step halt; fi
```

Comment:
With `[ -z "$CIRCLE_TAG" ]`, this *should* never exit early on tags. This is so that the deploy job can depend on PHPUnit tests and linting. 

Still, we might find out that doesn't work when actually tagging a release.

Reply:
Just noting that `$CIRCLE_TAG` can behave unexpectedly if repos are cached:
- https://discuss.circleci.com/t/fetching-circle-tag-doesnt-seem-to-work/19014
- https://discuss.circleci.com/t/builds-using-git-tags-failing/14904

Not a blocker because we're not caching git repos here as far as I'm aware: just noting in case we see it come up in the future.

Reply:
Oh, good point!